src/v/cluster/rm_stm.cc

```diff
@@ -858,8 +858,8 @@ ss::future<tx_errc> rm_stm::do_abort_tx(
             //
             // If it happens to be the first case then Redpanda rejects a
             // client's tx.
-            auto expiration_it = _mem_state.expiration.find(pid);
-            if (expiration_it != _mem_state.expiration.end()) {
+            auto expiration_it = _log_state.expiration.find(pid);
+            if (expiration_it != _log_state.expiration.end()) {
                 expiration_it->second.is_expiration_requested = true;
```

Comment:
perhaps you could combine this commit with the previous one which switches log_state and mem_state, and explain the transition in one commit. either way, both changes need some additional context added to the commit message.

Reply:
done